compiler-rt/lib/scudo/standalone/allocator_config.h

```diff
@@ -124,8 +124,14 @@ struct DefaultConfig {
   struct Primary {
     using SizeClassMap = DefaultSizeClassMap;
 #if SCUDO_CAN_USE_PRIMARY64
+#if SCUDO_RISCV64
+    // Support 39-bit VMA for riscv-64
+    static const uptr RegionSizeLog = 30U;
+    static const uptr GroupSizeLog = 19U;
+#else
```

Comment:
Can you share more detail about the failure you're seeing? Is the `RegionSizeLog = 30U` small enough for all 39-bit RISCV cases? I'm evaluating a different approach (#74531) to avoid the space preservation but I'm not sure if this solution will introduce a different fragmentation issue. If it's a different problem, I think we may want to have a complete different `DefaultConfig` for RISCV 39-bit (or having custom config for the specific project)

Reply:
`Scudo ERROR: internal map failure (NO MEMORY) requesting 188743680KB`

Reply:
Fuchsia has the default `RegionSizeLog=30` and `RegionSizeLog=28` for RISCV. There're other differences in the config but I want to know if `30` is smaller enough for RISCV or we should consider `28` as well. BTW, I think this is only for running tests, right?

Reply:
This is for running everything with Scudo. I found this problem when building Telegram Desktop on RISC-V, which switched to Scudo from Jemalloc some times ago.

Reply:
BTW at least I tested these values listed here can make telegram-desktop building process succeed.

Reply:
I see. Can we try 28 on your case and if that works, I would suggest having a slightly smaller number like that.

Reply:
well thus why is a smaller number suggested? shouldn't it be as big as the program do not crash?

Reply:
The reason for this crash is that the kernel can't reserve a contiguous pages which is big enough for this configuration. It pretty depends on how the VMA is managed in a process and how the libs are loaded. Besides, I suppose this change is only for tests. To avoid any other size change because of the same reason, a number as small as possible to pass the tests is more ideal.  If this is not only for the tests, then I would suggest having a custom config for your project instead.

BTW, as mentioned above, we will have a different approach for this issue and the change is only a workaround